cognite/client/data_classes/filters.py

```diff
@@ -202,6 +202,15 @@ def _involved_filter_types(self) -> set[type[Filter]]:
                 output.update(filter_._involved_filter_types())
         return output
 
+    def __and__(self, other: Filter) -> And:
+        return And(self, other)
```

Comment:
Consider checking if one is already of type `And`, and in such case, do not nest further `And`s, but merge 👌 

Same comment regarding `Or`